ovirt_imageio/client/_ovirt.py

```diff
@@ -425,6 +428,9 @@ def _wait_for_disk(con, disk_id):
         if time.monotonic() > deadline:
             raise RuntimeError(f"Timeout reached waiting for disk {disk_id}")
 
+        if disk_service.get().status == types.DiskStatus.ILLEGAL:
+            raise RuntimeError(f"Disk {disk_id} failed")
+
         if disk_service.get().status == types.DiskStatus.OK:
```

Comment:
Probably not important, but maybe we save the status once, and then compare it, instead of getting it multiple times, once for each comparison?

Reply:
Yeah I thought about it, but in the end it was one more line with an extra variable and decided otherwise. The reasoning was that this looked slightly better, and the performance is not that important in a wait loop.

But now that you've mentioned, I looked at it again, and I think you are right, so I probably change it lol

Reply:
I guess the extra requests matter more for the engine than the client.